third_party/libunwindstack/include/unwindstack/MapInfo.h

```diff
@@ -147,26 +151,29 @@ class MapInfo {
 
   inline bool IsBlank() { return offset() == 0 && flags() == 0 && name().empty(); }
 
-  // Returns elf_fields_. It will create the object if it is null.
-  ElfFields& GetElfFields();
+  // Returns object_fields_. It will create the object if it is null.
+  ObjectFields& GetObjectFields();
+
+  bool IsElf() const { return true; }
 
  private:
   MapInfo(const MapInfo&) = delete;
   void operator=(const MapInfo&) = delete;
 
   Memory* GetFileMemory();
   bool InitFileMemoryFromPreviousReadOnlyMap(MemoryFileAtOffset* memory);
+  Object* GetElf(const std::shared_ptr<Memory>& process_memory, ArchEnum expected_arch);
```

Comment:
Should this return a `Elf*`?

Reply:
Hmm, yeah, I guess it should, but then I think it looks up an `Object` in the cache. I need to take a closer look how to handle this.

Reply:
Feel free to postpone that. I guess you need to look at the details anyway when introducing `GetCOFF` later.

Reply:
I've changed it to basically be the same as what GetElf was before. Due to the interaction with the cache I think it's better to put the case distinction into the middle of GetObject. But what you say also makes a lot of sense, I'll have to revisit how this is structured when we actually have to init a Coff object here.